nearcore/src/config.rs

```diff
@@ -862,25 +888,29 @@ pub fn init_configs(
 ) -> anyhow::Result<()> {
     fs::create_dir_all(dir).with_context(|| anyhow!("Failed to create directory {:?}", dir))?;
 
+    let chain_id = chain_id
+        .and_then(|c| if c.is_empty() { None } else { Some(c.to_string()) })
+        .unwrap_or_else(random_chain_id);
+
     // Check if config already exists in home dir.
     if dir.join(CONFIG_FILENAME).exists() {
         let config = Config::from_file(&dir.join(CONFIG_FILENAME))
             .with_context(|| anyhow!("Failed to read config {}", dir.display()))?;
-        let file_path = dir.join(&config.genesis_file);
-        let genesis = GenesisConfig::from_file(&file_path).with_context(move || {
-            anyhow!("Failed to read genesis config {}/{}", dir.display(), config.genesis_file)
+        let genesis_file = config.genesis_file.clone();
+        let file_path = dir.join(&genesis_file);
+        // Check that Genesis exists and can be read.
```

Comment:
Can you explain the logic of the init function in comment in line 872?

so that we know what kind of scenarios it supports and which it doesn't.

as this code might require a little bit larger refactor.

Reply:
also - do we have any tests for this?

Reply:
Added comments.
Added a test.